### PR TITLE
feature(api): added api for faucet to allocate fund to user

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*

### CODE REVIEW

File: apps/server/src/faucet/FaucetController.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  constructor(
+    @Inject(CACHE_MANAGER) private cacheManager: any,
+    private readonly faucetService: FaucetService,
+    private configService: ConfigService,
+  ) {}
+
+  @Get(':address')
+  @UseInterceptors(AddressValidationInterceptor, CacheInterceptor, DefaultNetworkInterceptor)
+  async sendFunds(@Param('address') address: string, @Query('network') network: EnvironmentNetwork): Promise<TransactionResponse> {
+    const key = address;
```
Comment (lykalabrada) [inline apps/server/src/faucet/FaucetController.ts]: Nitpicking: Can we add a prefix for key? like `FAUCET_`

File: apps/server/src/faucet/FaucetController.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const isCached = await this.cacheManager.get(key);
+    if (isCached) {
+      throw new HttpException('Transfer already done, pleas try again later.', 403);
+    }
+    const amountToSend: string = this.configService.getOrThrow('faucetAmountPerRequest'); // Amount to send in DFI
+    const ttl = +this.configService.getOrThrow('faucetAmountPerRequest')
```
Comment (lykalabrada) [inline apps/server/src/faucet/FaucetController.ts]: Lets add a separate env variable for TTL, also add a comment if it's in sec or ms (or we can specify in the variable name itself, some suffix like `_MS`).
Thanks 🙏

**Review by lykalabrada — CHANGES_REQUESTED** [review]

File: apps/server/src/faucet/FaucetController.ts (hunk 2 above)
Comment (fullstackninja864) [reply apps/server/src/faucet/FaucetController.ts]: my bad we have already added env for this `throttleTimePerAddress`  env name `THROTTLE_TIME_PER_ADDRESS` all items are in seconds

File: apps/server/src/test/Faucet.spec.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    expect(txn?.status).toStrictEqual(1);
+  }, timeout);
+  await new Promise((resolve) => {
+    setTimeout(resolve, 1000);
+  });
+}
+
+describe('Faucet (e2e)', () => {
+  let app: NestFastifyApplication;
+  let defichain: StartedDeFiChainStubContainer;
```
Comment (chloezxyy) [inline apps/server/src/test/Faucet.spec.ts]: Nitpick: maybe can rename as `defichainContainer` instead?

**Review by lykalabrada — APPROVED** [review]

File: apps/server/src/faucet/FaucetController.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    private configService: ConfigService,
+  ) {}
+
+  @Get(':address')
+  @UseInterceptors(AddressValidationInterceptor, DefaultNetworkInterceptor)
+  async sendFunds(@Param('address') address: string, @Query('network') network: EnvironmentNetwork): Promise<TransactionResponse> {
+    const key = `FAUCET_${address}_${network}`;
+    const isCached = await this.cacheManager.get(key);
+    if (isCached) {
+      throw new HttpException('Transfer already done, pleas try again later.', 403);
```
Comment (pierregee) [inline apps/server/src/faucet/FaucetController.ts:26]:
```suggestion
      throw new HttpException('Transfer already done, please try again later.', 403);
```

**Review by pierregee — APPROVED** [review]

**Pull request merged**